src/libraries/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -473,7 +473,7 @@ public async Task SendMailAsync_CanBeCanceled_CancellationToken()
 
             var message = new MailMessage("foo@internet.com", "bar@internet.com", "Foo", "Bar");
 
-            Task sendTask = Task.Run(() => client.SendMailAsync(message, cts.Token));
+            Task sendTask = client.SendMailAsync(message, cts.Token);
```

Comment:
Why are there changes to the test?

Reply:
reverts #74910

Reply:
Why is that? If there is a reason to revert it, wouldn't it be better to do it in a separate PR?

Reply:
there is assumption that the issues are somewhat related. My changes were not meant as a fix but way how to make the test fail instead of hanging.

Reply:
> Why is that? If there is a reason to revert it, wouldn't it be better to do it in a separate PR?

Well, we can do it in anyway, but as @wfurt said, I also assumed they can be related. So, that's why I revert it in this PR.
If anybody thinks otherwise, I can revert this change and can push only fix.

Reply:
Well, I reverted it anyway. I'll create a seperate PR for this if necessary

Reply:
> Why is that?

We can imagine that if we're replacing handle that got disposed by one that is no longer disposed, that is causing a ODE to no longer be thrown somewhere. A future operation may get stuck instead of throwing.

It's not clear if this is actually what is happening. So it's good we've reverted the changes to the test.